lib/src/native/realm_core.dart

```diff
@@ -148,40 +147,231 @@ class _RealmCore {
+    final completer = Completer<SubscriptionSetState>();
+    _realmLib.realm_dart_sync_on_subscription_set_state_change_async(
+      subscriptions.handle._pointer,
+      notifyWhen.index,
+      Pointer.fromFunction(_stateChangeCallback),
+      completer.toPersistentHandle(),
+      _deletePersistentHandleFuncPtr,
+      subscriptions.realm.scheduler.handle._pointer,
+    );
+    return completer.future;
+  }
+
+  int subscriptionSetGetVersion(SubscriptionSet subscriptions) {
+    return _realmLib.realm_sync_subscription_set_version(subscriptions.handle._pointer);
+  }
+
+  SubscriptionSetState subscriptionSetGetState(SubscriptionSet subscriptions) {
+    return SubscriptionSetState.values[_realmLib.realm_sync_subscription_set_state(subscriptions.handle._pointer)];
+  }
+
+  MutableSubscriptionSetHandle subscriptionSetMakeMutable(SubscriptionSet subscriptions) {
+    return MutableSubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_make_subscription_set_mutable(subscriptions.handle._pointer)));
+  }
+
+  SubscriptionSetHandle subscriptionSetCommit(MutableSubscriptionSet subscriptions) {
+    return SubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_subscription_set_commit(subscriptions.handle._mutablePointer)));
+  }
+
+  SubscriptionHandle insertOrAssignSubscription(MutableSubscriptionSet subscriptions, RealmResults query, String? name, bool update) {
+    if (!update) {
+      if (name != null && findSubscriptionByName(subscriptions, name) != null) {
+        throw RealmException('Duplicate subscription $query${name != null ? ' with name: $name' : ''}');
       }
+    }
+    return using((arena) {
+      final out_index = arena<IntPtr>();
+      final out_inserted = arena<Uint8>();
+      _realmLib.invokeGetBool(() => _realmLib.realm_sync_subscription_set_insert_or_assign_results(
+            subscriptions.handle._mutablePointer,
+            query.handle._pointer,
+            name?.toUtf8Ptr(arena) ?? nullptr,
+            out_index,
+            out_inserted,
+          ));
+      return subscriptionAt(subscriptions, out_index.value);
+    });
+  }
 
-      return configHandle;
+  bool eraseSubscriptionById(MutableSubscriptionSet subscriptions, Subscription subscription) {
+    return using((arena) {
+      final out_found = arena.allocate<Uint8>(1);
+      _realmLib.invokeGetBool(() => _realmLib.realm_sync_subscription_set_erase_by_id(
+            subscriptions.handle._mutablePointer,
+            subscription.id.toCapi(arena),
+            out_found,
+          ));
+      return out_found.value != 0;
+    });
+  }
+
+  bool eraseSubscriptionByName(MutableSubscriptionSet subscriptions, String name) {
+    return using((arena) {
+      final out_found = arena.allocate<Uint8>(1);
+      _realmLib.invokeGetBool(() => _realmLib.realm_sync_subscription_set_erase_by_name(
+            subscriptions.handle._mutablePointer,
```

Comment:
I need both.

Reply:
Maybe I don't ready it right, but we are initializing the `super._pointer`  with the same value. Than we are introducing new name `_mutablePointer` which reads the `super._pointer`. Why not read `_pointer` and not introduce new name here?

Reply:
Because the type differs. Otherwise, I need to cast on all call-sites, ie. `subscription.handle._pointer.cast()`. That will also work. Do you prefer that?